starboard/rdk/launcher.py

```diff
@@ -0,0 +1,426 @@
+        backoff=lambda: self.shutdown_initiated.is_set(),
+        wrap_exceptions=False)
+    def _readloop():
+      test_completed = 0
+      while True:
+        # Sanitize the line to remove ansi color codes.
+        line = Launcher._PEXPECT_SANITIZE_LINE_RE.sub(
+            '', self.pexpect_process.readline())
+        self.output_file.flush()
+        if not line:
+          return
+        # Check for the test complete tag. It will be followed by either a
+        # success or failure tag.
+        if line.find(self.test_complete_tag) != -1:
+          test_completed = 1
+        if test_completed and line.find(self.test_failure_tag) != -1:
+          self.return_value = 1
+        if self.return_value and line.find(self.test_failure_list_tag) != -1:
+          return
+
+    _readloop()
+
+  def _Sleep(self, val):
+    self._PexpectSendLine(f'sleep {val};echo {Launcher._SSH_SLEEP_SIGNAL}')
+    self.pexpect_process.expect([Launcher._SSH_SLEEP_SIGNAL])
+
+  def _CleanupPexpectProcess(self):
+    """Closes current pexpect process."""
+
+    if self.pexpect_process is not None and self.pexpect_process.isalive():
+      # Check if kernel logged OOM kill or any other system failure message
+      if self.return_value:
+        logging.info('Sending dmesg')
+        with contextlib.suppress(Launcher._RETRY_EXCEPTIONS):
+          self._PexpectSendLine('dmesg -P --color=never | tail -n 100')
+        time.sleep(self._PEXPECT_SHUTDOWN_SLEEP_TIME)
+        with contextlib.suppress(Launcher._RETRY_EXCEPTIONS):
+          self.pexpect_process.readlines()
+        logging.info('Done sending dmesg')
+
+      # Send ctrl-c to the RDK and close the process.
+      with contextlib.suppress(Launcher._RETRY_EXCEPTIONS):
+        self._PexpectSendLine(chr(3))
+      time.sleep(self._PEXPECT_TIMEOUT)  # Allow time for normal shutdown
+      with contextlib.suppress(Launcher._RETRY_EXCEPTIONS):
+        self.pexpect_process.close()
+
+  def _WaitForPrompt(self):
+    """Sends empty commands, until a bash prompt is returned"""
+
+    def backoff():
+      self._PexpectSendLine('echo ' + Launcher._SSH_SLEEP_SIGNAL)
+      return self._ShutdownBackoff()
+
+    retry.with_retry(
+        lambda: self.pexpect_process.expect(self._RDK_PROMPT),
+        exceptions=Launcher._RETRY_EXCEPTIONS,
+        retries=Launcher._PROMPT_WAIT_MAX_RETRIES,
+        backoff=backoff,
+        wrap_exceptions=False)
+
+  @retry.retry(
+      exceptions=_RETRY_EXCEPTIONS,
+      retries=_KILL_RETRIES,
+      backoff=_CommandBackoff)
+  def _KillExistingCobaltProcesses(self):
+    """If there are leftover Cobalt processes, kill them.
+
+    It is possible that a previous process did not exit cleanly.
+    Zombie Cobalt instances can block the WebDriver port or
+    cause other problems.
+    """
+    logging.info('Killing existing processes')
+    self._PexpectSendLine(
+        'pkill -9 -ef "(cobalt)|(crashpad_handler)|(elf_loader)"')
```

Comment:
Does pkill work on RDK? In the 'Run local build Evergreen on RDK' doc you mention a terminate command. Should that be used instead?